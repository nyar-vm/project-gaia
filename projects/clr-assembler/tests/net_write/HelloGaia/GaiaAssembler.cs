using System;

namespace Valkyrie.Translator.HelloGaia;

public class GaiaAssembler
{
    public static void Main(string[] args)
    {
        Console.WriteLine("Hello Gaia!");
    }
}
