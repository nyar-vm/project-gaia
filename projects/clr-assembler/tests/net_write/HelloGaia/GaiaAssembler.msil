.assembly extern mscorlib
{
  .publickeytoken = (B7 7A 5C 56 19 34 E0 89 )
  .ver 4:0:0:0
}

.assembly extern System.Console
{
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 4:0:0:0
}

.assembly GaiaAssembler
{
  .hash algorithm 0x00008004
  .ver 0:0:0:0
}

.module GaiaAssembler.exe

.class public auto ansi beforefieldinit
  Valkyrie.Translator.HelloGaia.GaiaAssembler extends [mscorlib]System.Object
{
  .method public hidebysig static void Main(string[] args) cil managed
  {
    .entrypoint
    .maxstack 8

    ldstr        "Hello Gaia!"
    call         void [System.Console]System.Console::WriteLine(string)
    nop
    ret
  }

  .method public hidebysig specialname rtspecialname instance void
    .ctor() cil managed
  {
    .maxstack 8

    ldarg.0      // this
    call         instance void [mscorlib]System.Object::.ctor()
    nop
    ret
  }
}