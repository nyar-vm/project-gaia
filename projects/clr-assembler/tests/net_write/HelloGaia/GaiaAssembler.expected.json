{
    "assembly_name": "GaiaAssembler",
    "extern_assemblies": [
        "mscorlib",
        "System"
    ],
    "module_name": "GaiaAssembler",
    "class_name": "Valkyrie.Translator.HelloGaia.GaiaAssembler",
    "class_modifiers": [
        "public",
        "auto",
        "ansi",
        "beforefieldinit"
    ],
    "extends": "[mscorlib]System.Object",
    "method_count": 2,
    "method_names": [
        "Main",
        ".ctor"
    ],
    "statement_count": 5
}